src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayExtensions.cs

```diff
@@ -54,6 +54,46 @@ public static string ToDisplayString(this ImmutableArray<SymbolDisplayPart> part
             }
         }
 
+        /// <summary>
+        /// Converts an ArrayBuilder of <see cref="SymbolDisplayPart"/>s to a string.
+        /// </summary>
+        /// <param name="parts">The array of parts.</param>
+        /// <returns>The concatenation of the parts into a single string.</returns>
+        [PerformanceSensitive("https://github.com/dotnet/roslyn/pull/67203", AllowImplicitBoxing = false)]
+        internal static string ToDisplayString(this ArrayBuilder<SymbolDisplayPart> parts)
+        {
+            if (parts is null)
+            {
+                throw new ArgumentException("parts");
+            }
+
+            if (parts.Count == 0)
+            {
+                return string.Empty;
+            }
+
+            if (parts.Count == 1)
+            {
+                return parts[0].ToString();
+            }
```

Comment:
```suggestion
            if (parts is [var singlePart])
            {
                return singlePart.ToString();
            }
```

Reply:
I'm a bit hesitant to make this change. I'm not of the opinion that it improves readability in this case, and I *think* it might add a null check in the code.